From 625dea43c80dc7f25223e6c9239ca1322d07ab2c Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Thu, 12 Sep 2024 17:34:59 +0100
Subject: [PATCH 3/3] ZZZgnutls-build.patch: De-conditionalize libcurlu.la
 parameters

The static library used to link unit tests is always libcurlu.la, never
libcurlu-gnutls.la, regardless of whether the shared library is going
to be libcurl.la (OpenSSL flavour) or libcurl-gnutls.la (GNUTLS flavour).
---
 debian/patches/ZZZgnutls-build.patch | 20 +++++++-------------
 1 file changed, 7 insertions(+), 13 deletions(-)

diff --git a/debian/patches/ZZZgnutls-build.patch b/debian/patches/ZZZgnutls-build.patch
index 85e2ccca12..f403a83aea 100644
--- a/debian/patches/ZZZgnutls-build.patch
+++ b/debian/patches/ZZZgnutls-build.patch
@@ -8,13 +8,13 @@ Last-Update: 2024-09-12
 ---
  configure.ac                   | 10 ++++++++
  docs/examples/Makefile.am      |  2 +-
- lib/Makefile.am                | 56 ++++++++++++++++++++++++++++++++++++++++--
+ lib/Makefile.am                | 53 ++++++++++++++++++++++++++++++++++++++++--
  lib/libcurl.vers.in            |  2 +-
  src/Makefile.am                |  2 +-
  tests/http/clients/Makefile.am |  2 +-
  tests/http/clients/Makefile.in | 18 +++++++-------
- tests/libtest/Makefile.am      |  4 +--
- 8 files changed, 79 insertions(+), 17 deletions(-)
+ tests/libtest/Makefile.am      |  4 ++--
+ 8 files changed, 76 insertions(+), 17 deletions(-)
 
 diff --git a/configure.ac b/configure.ac
 index 269d7bd..b87b7cf 100644
@@ -51,7 +51,7 @@ index 91f90cf..13874e1 100644
  # This might hold -Werror
  CFLAGS += @CURL_CFLAG_EXTRAS@
 diff --git a/lib/Makefile.am b/lib/Makefile.am
-index 851cee2..f0fddca 100644
+index 851cee2..2674a22 100644
 --- a/lib/Makefile.am
 +++ b/lib/Makefile.am
 @@ -34,7 +34,11 @@ EXTRA_DIST = Makefile.mk config-win32.h config-win32ce.h config-plan9.h \
@@ -66,13 +66,13 @@ index 851cee2..f0fddca 100644
  
  if BUILD_UNITTESTS
  noinst_LTLIBRARIES = libcurlu.la
-@@ -70,55 +74,103 @@ AM_CFLAGS =
+@@ -70,52 +74,97 @@ AM_CFLAGS =
  # Makefile.inc provides the CSOURCES and HHEADERS defines
  include Makefile.inc
  
 +if GNUTLSBUILD
 +libcurl_gnutls_la_SOURCES = $(CSOURCES) $(HHEADERS)
-+libcurlu_gnutls_la_SOURCES = $(CSOURCES) $(HHEADERS)
++libcurlu_la_SOURCES = $(CSOURCES) $(HHEADERS)
 +libcurl_gnutls_la_CPPFLAGS_EXTRA =
 +libcurl_gnutls_la_LDFLAGS_EXTRA =
 +libcurl_gnutls_la_CFLAGS_EXTRA =
@@ -157,21 +157,15 @@ index 851cee2..f0fddca 100644
 +libcurl_gnutls_la_CPPFLAGS = $(AM_CPPFLAGS) $(libcurl_gnutls_la_CPPFLAGS_EXTRA)
 +libcurl_gnutls_la_LDFLAGS = $(AM_LDFLAGS) $(libcurl_gnutls_la_LDFLAGS_EXTRA) $(CURL_LDFLAGS_LIB) $(LIBCURL_PC_LIBS_PRIVATE)
 +libcurl_gnutls_la_CFLAGS = $(AM_CFLAGS) $(libcurl_gnutls_la_CFLAGS_EXTRA)
-+libcurlu_gnutls_la_CPPFLAGS = $(AM_CPPFLAGS) -DCURL_STATICLIB -DUNITTESTS
-+libcurlu_gnutls_la_LDFLAGS = $(AM_LDFLAGS) -static $(LIBCURL_LIBS)
-+libcurlu_gnutls_la_CFLAGS = $(AM_CFLAGS)
 +else
  libcurl_la_CPPFLAGS = $(AM_CPPFLAGS) $(libcurl_la_CPPFLAGS_EXTRA)
  libcurl_la_LDFLAGS = $(AM_LDFLAGS) $(libcurl_la_LDFLAGS_EXTRA) $(CURL_LDFLAGS_LIB) $(LIBCURL_PC_LIBS_PRIVATE)
  libcurl_la_CFLAGS = $(AM_CFLAGS) $(libcurl_la_CFLAGS_EXTRA)
 -
++endif
  libcurlu_la_CPPFLAGS = $(AM_CPPFLAGS) -DCURL_STATICLIB -DUNITTESTS
  libcurlu_la_LDFLAGS = $(AM_LDFLAGS) -static $(LIBCURL_PC_LIBS_PRIVATE)
  libcurlu_la_CFLAGS = $(AM_CFLAGS)
-+endif
- 
- CHECKSRC = $(CS_$(V))
- CS_0 = @echo "  RUN     " $@;
 diff --git a/lib/libcurl.vers.in b/lib/libcurl.vers.in
 index ae978a4..7fd4c07 100644
 --- a/lib/libcurl.vers.in
-- 
2.45.2

